src/query/functions/aggregation/take.go

```diff
@@ -162,19 +182,58 @@ func (n *takeNode) ProcessBlock(queryCtx *models.QueryContext, ID parser.NodeID,
 	return builder.Build(), nil
 }
 
+func (n *takeNode) appendValuesInstant(builder block.Builder, stepIter block.StepIter, seriesMetas []block.SeriesMeta, buckets [][]int) error {
+	stepCount := stepIter.StepCount()
+	for index := 0; stepIter.Next(); index++ {
+		values := stepIter.Current().Values()
+
+		if err := builder.AppendValues(index, values); err != nil {
+			return err
+		}
```

Comment:
Should this be appending previous values? This would probably be cleaner if `func (o takeOp) Node` yields either a `takeOp` or an `instantTakeOp`, where `instantTakeOp` performs the block building itself, rather than trying to write a bunch of values, then overwrite them with `builder.SetRow`

Reply:
Agree. Actually, I was trying to do block building itself but was not sure how to build a block of a single column without getting an error.